frontend/pages/campaigns/[campaignId].tsx

```diff
@@ -0,0 +1,102 @@
+import { useRouter } from 'next/router';
+import React from 'react';
+import { canBecomeInteger } from '../../utils/numbers';
+import CampaignCharityCard from '../../components/charities/CampaignCharityCard';
+import { Box, Button, Divider, Stack, Typography } from '@mui/material';
+import { theme } from '../../utils/theme';
+import CircularProgressWithLabel from '../../components/CircularProgressWithLabel';
+import SwiperWrapper from '../../components/swiper/SwiperWrapper';
+import CampaignCharityList from '../../components/charities/CampaignCharityList';
+import api from '../../frontendApis';
+import { GetServerSidePropsContext } from 'next';
+import { CampaignPublicData } from '../../types/campaigns';
+import { StatusMessage } from '../../types/api';
+import { Nullable } from '../../types/utils';
+import { useSnackbar } from 'notistack';
+import { enqueueGCSnackbar } from '../../utils/snackbar';
+import CampaignDescription from '../../components/campaigns/CampaignDescription';
+
+export async function getServerSideProps(
+  context: GetServerSidePropsContext,
+): Promise<{ props: { campaign: Nullable<CampaignPublicData>; message?: Nullable<StatusMessage> } }> {
+  const { campaignId: campaignIdParam } = context.query;
+
+  const campaignId = canBecomeInteger(campaignIdParam) ? Number(campaignIdParam) : undefined;
+
+  if (!campaignId) {
+    return {
+      props: {
+        campaign: null,
+      },
+    };
+  }
+
+  const response = await api.campaigns.getCampaign(campaignId);
```

Comment:
<img width="1431" alt="Screenshot 2022-10-16 at 1 02 15 PM" src="https://user-images.githubusercontent.com/71375383/196018982-ebcc1eb1-61ac-4a6f-86df-428dd15e9049.png">

There's a runtime error for not found campaigns. Actually why are we not using swr here?

Reply:
I think we decided to use SSR since the request does not depend on state of the client so that we get better initial load times and SEO.

Reply:
Just realised that SSR does not work since we are not doing a next serverless deployment. Will change to SWR